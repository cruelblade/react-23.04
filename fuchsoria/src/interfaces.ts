export interface IMessage {
  id: string;
  author?: string;
  text: string;
  authorAccess?: string;
  isSelf?: boolean;
  date?: number;
}

export interface IMessageWithHandler extends IMessage {
  handleDelete: () => void;
}

export interface IChatListItem {
  id: number | string;
  title: string;
  description: string;
}

export interface IChatListProps {
  items: IChatListItem[];
  blinkingIds?: string[];
}

export interface IChatContainerProps {
  chats: IChats;
  profile: IProfileState;
  addChat: (chatName: string, chatId: string) => void;
  addMessage: (message: IMessage, chatId: string, callback?: Function) => void;
  noContent?: boolean;
  history?: unknown;
  location?: unknown;
  match?: {
    isExact: boolean;
    params: {
      chatId: string;
    };
    path: string;
    url: string;
  };
}

export interface IProfileState {
  nickName: string;
}

export interface IChatContainerState {
  chatList?: IChatListItem[];
}

export interface IMessageListProps {
  messages: IMessage[];
  chatId: string;
}

export interface IChatFormProps {
  handleSubmit: (text: string, author?: string) => void;
}

export interface IChat {
  title: string;
  messages: IMessage[];
}

export interface IChats {
  [key: string]: IChat;
}

export interface IUpdateProfile {
  nickName?: string;
}

export interface IProfileFormProps {
  updateProfile: ({ nickName }: IUpdateProfile) => void;
}
