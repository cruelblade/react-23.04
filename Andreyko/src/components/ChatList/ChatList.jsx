import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { List, ListItem } from 'material-ui/List';
import './style.css';

export default class ChatList extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <List>
        <Link to="/chat/1/">
          <ListItem primaryText="Игорь" />
        </Link>
        <Link to="/chat/2/">
          <ListItem primaryText="Ламповый Ром4нтик" />
        </Link>
        <Link to="/chat/3/">
          <ListItem primaryText="Тайный покупатель" />
        </Link>
      </List>
    )
  }
}