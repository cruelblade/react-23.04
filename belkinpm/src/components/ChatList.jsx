import React, { Component } from 'react';

class ChatList extends Component {
  render() {
    return (
        <div className="chat-list">
            <ul>
                <li>John Smith</li>
                <li>Donny Trump</li>
                <li className="active">Robot Rock</li>
                <li>Marry Poppins</li>
                <li>Santa</li>
            </ul>
        </div>
    )
  }
}

export default ChatList;
