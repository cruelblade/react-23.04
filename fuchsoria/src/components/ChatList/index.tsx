import React from 'react';
import { List, Avatar } from 'antd';
import { useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { IChatListProps } from '../../interfaces';
import { State } from '../../store/reducers/reducerTypes';
import { deleteChat } from '../../store/actions/chatsActions';
import styles from './styles.module.scss';

export default function ChatList({ items }: IChatListProps) {
  const blinkingIds = useSelector((store: State) => store.chatlist.blinkingIds);
  const history = useHistory();
  const dispatch = useDispatch();

  const switchChat = (chatId: string) => () => {
    history.push(`/chats/${chatId}`);
  };

  const handleDelete = (chatId: string) => () => {
    const isConfirmed = confirm('Remove chat?');

    if (isConfirmed) {
      dispatch(deleteChat(chatId));

      setTimeout(() => history.push('/chats/'), 300);
    }
  };

  return (
    <List
      className={styles.chatList}
      itemLayout="horizontal"
      dataSource={items}
      renderItem={(item) => (
        <List.Item
          className={styles.chatListItem}
          onClick={switchChat(String(item.id))}
          onDoubleClick={handleDelete(String(item.id))}
        >
          <List.Item.Meta
            avatar={<Avatar src="https://source.unsplash.com/random/300x300" />}
            title={item.title}
            description={blinkingIds.includes(String(item.id)) ? '[NEW ROBOT MSG]' : item.description}
          />
        </List.Item>
      )}
    />
  );
}
