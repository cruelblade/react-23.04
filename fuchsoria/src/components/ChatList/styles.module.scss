.chat-list {
  padding: 10px;
  overflow-y: scroll;
  height: 50vh;

  &__item {
    user-select: none;
    border-bottom: 1px solid #d6d6d6 !important;
    cursor: pointer;
    transition: opacity 0.3s;

    &:hover {
      opacity: 0.7;
    }
  }
}
