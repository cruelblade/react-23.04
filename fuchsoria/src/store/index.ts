import { fetchMiddleware } from './middlewares/fetchMiddleware';
import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import { composeWithDevTools } from 'redux-devtools-extension';
import storage from 'redux-persist/lib/storage';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { apiMiddleware } from 'redux-api-middleware';
import chatsReducer from './reducers/chatsReducer';
import profileReducer from './reducers/profileReducer';
import chatlistReducer from './reducers/chatlistReducer';
import { botAnswer } from './middlewares/botMiddleware';
import { autoRedirect } from './middlewares/chatMiddleware';
import { createBrowserHistory, History } from 'history';

const history = createBrowserHistory();

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const reducer = (history: History) =>
  combineReducers({
    chats: chatsReducer,
    profile: profileReducer,
    chatlist: chatlistReducer,
    router: connectRouter(history),
  });

const persistConfig = {
  key: 'messengerweb',
  whitelist: ['chats', 'profile'],
  storage,
};

export function initStore(preloadedState = undefined) {
  const persistedReducer = persistReducer(persistConfig, reducer(history));

  const middlewares = [routerMiddleware(history), apiMiddleware, fetchMiddleware, thunk, botAnswer, autoRedirect];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(persistedReducer, preloadedState, composedEnhancers);
  const persistor = persistStore(store);

  return { store, persistor, history };
}
