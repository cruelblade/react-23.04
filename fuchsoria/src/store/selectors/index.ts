import { IChats, IProfileState } from './../../interfaces';

export const selectChats = (state: { chats: IChats }) => {
  const { chats } = state;

  return chats;
};

export const selectProfile = (state: { profile: IProfileState }) => {
  const { profile } = state;

  return profile;
};
