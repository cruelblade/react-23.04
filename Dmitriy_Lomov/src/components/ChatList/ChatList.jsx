import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { getAllChats, getChatUpdatedIds } from '../../store/chats/selectors';
import PropTypes from 'prop-types';

import styles from './index.css';
import { withStyles } from '@material-ui/core/styles';
import cx from 'classnames';
import {
  Drawer,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Typography,
} from '@material-ui/core';
import ChatIcon from '@material-ui/icons/Chat';
import HomeIcon from '@material-ui/icons/Home';

const drawerWidth = 240;

const muiStyles = theme => ({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
    whiteSpace: 'nowrap',
  },

  logo: {
    marginRight: 'auto',
    marginLeft: theme.spacing(1),
  },

  paper: {
    color: theme.palette.text.primary,
  },

  drawerOpen: {
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },

  itemRoot: {
    position: 'relative',
    whiteSpace: 'pre-line',
  },

  link: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    zIndex: theme.zIndex.appBar,
  },

  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: theme.spacing(3, 2),
  },

  toolbarClose: {
    justifyContent: 'flex-start',
  },
});

class ChatList extends PureComponent {
  render() {
    const { chats, classes, chatUpdatedIds } = this.props;
    return (
      <Drawer
        variant="permanent"
        className={cx(classes.drawer, classes.drawerOpen)}
        classes={{
          paper: cx(classes.paper, classes.drawerOpen),
        }}
      >
        <div className={classes.toolbar}>
          <Typography variant="body1">Список чатов</Typography>
        </div>
        <Link to="/" key="home-page">
          <ListItem
            classes={{
              root: classes.itemRoot,
            }}
            button
          >
            <ListItemIcon>
              <HomeIcon />
            </ListItemIcon>
            <ListItemText>Home</ListItemText>
          </ListItem>
        </Link>
        <List disablePadding>
          {chats.map(({ id, title, to }) => (
            <Link to={to} key={id}>
              <ListItem
                className={cx({ [styles.highlighter]: chatUpdatedIds.includes(id) })}
                classes={{
                  root: classes.itemRoot,
                }}
                button
              >
                <ListItemIcon>
                  <ChatIcon />
                </ListItemIcon>
                <ListItemText>{title}</ListItemText>
              </ListItem>
            </Link>
          ))}
        </List>
      </Drawer>
    );
  }
}

ChatList.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = store => ({
  chats: getAllChats(store),
  chatUpdatedIds: getChatUpdatedIds(store),
});

export default compose(connect(mapStateToProps), withStyles(muiStyles))(ChatList);
