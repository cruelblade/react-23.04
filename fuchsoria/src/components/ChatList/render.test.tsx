import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initStore } from '../../store';
import { IChatListProps } from '../../interfaces';
import ChatList from '../ChatList';

const { store } = initStore();

const mockData: IChatListProps = {
  items: [
    {
      id: 0,
      title: 'John',
      description: 'What are you doing?',
    },
    {
      id: 1,
      title: 'Heinz',
      description: 'What are you doing?',
    },
    {
      id: 2,
      title: 'Dungeon Master',
      description: 'What are you doing?',
    },
  ],
  blinkingIds: [],
};

describe('<ChatList/>', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => {
        return {
          matches: true,
          addListener: jest.fn(),
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
        };
      }),
    });
  });

  it('Component should render', () => {
    const { container } = render(
      <Provider store={store}>
        <ChatList items={mockData.items} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
