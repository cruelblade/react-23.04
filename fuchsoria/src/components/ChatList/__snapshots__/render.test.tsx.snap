// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList/> Component should render 1`] = `
<div>
  <div
    class="ant-list chatList ant-list-split"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <ul
          class="ant-list-items"
        >
          <li
            class="ant-list-item chatListItem"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-avatar"
              >
                <span
                  class="ant-avatar ant-avatar-circle ant-avatar-image"
                >
                  <img
                    src="https://source.unsplash.com/random/300x300"
                  />
                </span>
              </div>
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  John
                </h4>
                <div
                  class="ant-list-item-meta-description"
                >
                  What are you doing?
                </div>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item chatListItem"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-avatar"
              >
                <span
                  class="ant-avatar ant-avatar-circle ant-avatar-image"
                >
                  <img
                    src="https://source.unsplash.com/random/300x300"
                  />
                </span>
              </div>
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  Heinz
                </h4>
                <div
                  class="ant-list-item-meta-description"
                >
                  What are you doing?
                </div>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item chatListItem"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-avatar"
              >
                <span
                  class="ant-avatar ant-avatar-circle ant-avatar-image"
                >
                  <img
                    src="https://source.unsplash.com/random/300x300"
                  />
                </span>
              </div>
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  Dungeon Master
                </h4>
                <div
                  class="ant-list-item-meta-description"
                >
                  What are you doing?
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
